app/models/user.rb

```diff
@@ -12,5 +12,12 @@ class User < ApplicationRecord
   has_many :activities, dependent: :destroy
   has_many :notifications, dependent: :destroy
 
-  groupify :group_member
+  has_many :group_memberships
+  has_many :groups, through: :group_memberships
+
+  def in_group?(group, opts = {})
+    return false unless group.present?
+
+    groups.as(opts[:as]).include?(group)
```

Comment:
that's kinda weird that we use the scope when `as` is optional, prob should be `opts[:as] ? ... : ..`

Reply:
if you move the scope to membership, than you can avoid loading all groups as in `group_memberships.as(opts[:as]).where(group: group).exists?`